File: FarmProject.React/src/types/Customer.ts
export interface Order {
  id: number;
  customerId: number;
  orderStatus: number;
  orderDate: string;
}

export interface Customer {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNum: string;
}

export interface CustomerWithOrders {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNum: string;
  orders: Order[];
}

export interface CustomerResponse {
  pageIndex: number;
  pageSize: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
  items: Customer[];
}

File: FarmProject.Domain/Models/Customer.cs
namespace FarmProject.Domain.Models;

public class Customer(
        string firstName,
        string lastName,
        string email,
        string phoneNum) 
    : Entity
{
    public string FirstName { get; private set; } = firstName;
    public string LastName { get; private set; } = lastName;
    public string Email { get; private set; } = email;
    public string PhoneNum { get; private set; } = phoneNum;
    public List<Order> Orders { get; private set; } = new();
}
